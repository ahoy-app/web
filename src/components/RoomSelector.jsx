import React from 'react'
import Radium from 'radium'
import PropTypes from 'prop-types'

import RoomBubble from './RoomBubble'
import Spacer from './Spacer'

import { color } from '../style'

const styles = {
  background: color.main.dark,
  height: '100%',
  width: '45px',

  padding: '0px 5px 0px 5px',

  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'flex-start',
  alignItems: 'center',
}

const RoomSelector = ({ rooms }) => (
  <div style={styles}>
    {rooms.map(room => (
      <Spacer top={5} bottom={5} key={room.id}>
        <RoomBubble room={room} />
      </Spacer>
    ))}
  </div>
)

RoomSelector.propTypes = {
  rooms: PropTypes.array.isRequired,
}

export default Radium(RoomSelector)
