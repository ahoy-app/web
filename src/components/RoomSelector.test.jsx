import React from 'react'
import { mount } from 'enzyme'

import RoomSelector from './RoomSelector'

const rooms = ['mclarens', 'GNB']

let component

describe('RoomSelector', () => {
  afterAll(() => {
    if (component) component.unmount()
  })
  it('should render correctly', () => {
    component = mount(<RoomSelector rooms={rooms} />)
    expect(component).toMatchSnapshot()
  })
})
