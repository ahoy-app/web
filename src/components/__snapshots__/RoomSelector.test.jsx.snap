// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomSelector should render correctly 1`] = `
<RoomSelector
  rooms={
    Array [
      "mclarens",
      "GNB",
    ]
  }
>
  <div
    data-radium={true}
    style={
      Object {
        "alignItems": "center",
        "background": "linear-gradient(45deg, #045 0%, #9bb 100%)",
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "flex-start",
        "padding": "0px 5px 0px 5px",
        "width": "45px",
      }
    }
  >
    <Spacer
      bottom={5}
      key=".$mclarens"
      top={5}
    >
      <div
        data-radium={true}
        style={
          Object {
            "padding": "5px 0px 5px 0px",
          }
        }
      >
        <RoomBubble
          room="mclarens"
        >
          <div
            data-radium={true}
            style={
              Object {
                "backgroundColor": "gray",
                "borderRadius": "50%",
                "height": "40px",
                "lineHeight": "40px",
                "textAlign": "center",
                "width": "40px",
              }
            }
          >
            M
          </div>
        </RoomBubble>
      </div>
    </Spacer>
    <Spacer
      bottom={5}
      key=".$GNB"
      top={5}
    >
      <div
        data-radium={true}
        style={
          Object {
            "padding": "5px 0px 5px 0px",
          }
        }
      >
        <RoomBubble
          room="GNB"
        >
          <div
            data-radium={true}
            style={
              Object {
                "backgroundColor": "gray",
                "borderRadius": "50%",
                "height": "40px",
                "lineHeight": "40px",
                "textAlign": "center",
                "width": "40px",
              }
            }
          >
            G
          </div>
        </RoomBubble>
      </div>
    </Spacer>
  </div>
</RoomSelector>
`;
